### ISSUE TITLE
Suggest tuple struct syntax when using struct incorrectly

### ISSUE THREAD

**Issue by estebank** [rust-lang/rust#57242]
As per https://github.com/rust-lang/rust/pull/57047#discussion_r244608381, when encountering a tuple struct being used with the wrong syntax, suggest an appropriate fix:

```
error[E0560]: struct `NonCopyable` has no field named `p`
  --> $DIR/issue-4736.rs:4:26
   |
LL |     let z = NonCopyable{ p: () }; //~ ERROR struct `NonCopyable` has no field named `p`
   |             -------------^-----
   |             |            |
   |             |            field does not exist
   |             help: `NonCopyable` is a tuple struct, use the appropriate syntax: `NonCopyable(())`
```

instead of

```
error[E0560]: struct `NonCopyable` has no field named `p`
  --> $DIR/issue-4736.rs:4:26
   |
LL |     let z = NonCopyable{ p: () }; //~ ERROR struct `NonCopyable` has no field named `p`
   |                          ^ help: a field with a similar name exists: `0`
```

**Comment by Electron-libre** [comment]
Hi, 

I can give it a try. This does not seems hard to fix.

Probably from here: https://github.com/rust-lang/rust/blob/8e948df707ea8a3c88c65bf2ffdcb2f1cf5491be/src/librustc_typeck/check/mod.rs#L3622

+ 1 Unit test

Should be enought

**Comment by estebank** [comment]
@Electron-libre that'd be great!

You'll have to look at the `AdtDef`'s (only) variant in `variants`, which is of type `VariantDef`, and look at its `ctor_kind` field (of type `CtorKind`) to identify whether this is a struct or tuple style struct. From there it is a matter of using the appropriate spans for the suggestion and silencing the "similar name exists" suggestion.

**Comment by Electron-libre** [comment]
Thanks for the tips.

**Comment by JohnTitor** [comment]
@estebank this can be closed? It seems fixed in #61782.
